pkg/splunk/client/azureblobclient.go

```diff
@@ -108,126 +207,242 @@ func NewAzureBlobClient(ctx context.Context, bucketName string, storageAccountNa
-	// httpClient := http.Client{Transport: tr}
+// Update http request header with IAM info
+func updateAzureHTTPRequestHeaderWithIAM(ctx context.Context, client *AzureBlobClient, httpRequest *http.Request) error {
+	reqLogger := log.FromContext(ctx)
+	scopedLog := reqLogger.WithName("updateHttpRequestHeaderWithIAM")
 
-	// config := &Config{
-	// 	MaxRetries: 3,
-	// 	HTTPClient: httpClient,
-	// }
+	scopedLog.Info("Updating Azure Http Request with IAM")
 
-	// azureBlobClient, err = NewAzureSession(ctx, appAzureBlobEndPoint, storageAccountName, secretAccessKey, config)
-	// if err != nil {
-	// 	scopedLog.Info("Error creating new AzureBlob Client Session", "err", err)
-	// 	return nil
-	// }
+	// Create http request to retrive IAM oauth token from metadata URL
+	oauthRequest, err := http.NewRequest("GET", azureTokenFetchURL, nil)
+	if err != nil {
+		scopedLog.Error(err, "Azure Blob Failed to create new token request")
+		return err
+	}
 
-	azureBlobClient = SplunkAzureBlobClientTemp{client: http.Client{}}
-	return &azureBlobClient
+	// Mark metadata flag
+	oauthRequest.Header.Set("Metadata", "true")
 
-}
+	// Create raw query for http request
+	values := oauthRequest.URL.Query()
+	values.Add("api-version", "2018-02-01")
+	values.Add("resource", "https://storage.azure.com/")
+	oauthRequest.URL.RawQuery = values.Encode()
 
-// GetAppsList get the list of apps from remote storage
-func (client *AzureBlobClient) GetAppsList(ctx context.Context) (RemoteDataListResponse, error) {
-	reqLogger := log.FromContext(ctx)
-	scopedLog := reqLogger.WithName("GetAppsList")
+	// Retrieve oauth token
+	resp, err := client.HTTPClient.Do(oauthRequest)
+	if err != nil {
+		scopedLog.Error(err, "Azure blob,Errored when sending request to the server")
+		return err
+	}
+	defer resp.Body.Close()
 
-	scopedLog.Info("Getting Apps list", " S3 Bucket", client.BucketName, "Prefix", client.Prefix)
+	// Read http response
+	responseBody, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		scopedLog.Error(err, "Azure blob,Errored when reading resp body")
+		return err
+	}
 
-	azureBlobClient := client.Client
+	// Extract the token from the http response
+	var azureOauthTokenResponse TokenResponse
+	err = json.Unmarshal(responseBody, &azureOauthTokenResponse)
+	if err != nil {
+		scopedLog.Error(err, "Unable to unmarshal response to token")
+		return err
+	}
 
-	// Create a bucket list command for all files in bucket
+	// Update http request header with IAM access token
+	httpRequest.Header.Set("x-ms-version", azureHTTPHeaderXmsVersion)
+	httpRequest.Header.Set("Authorization", "Bearer "+azureOauthTokenResponse.AccessToken)
 
-	//TODO: revise it in next sprint. had to use map temporary to let build work
-	opts := make(map[string]string)
+	return nil
+}
 
-	opts["Prefix"] = client.Prefix
-	opts["MaxEnteries"] = "4000"
+// GetAppsList gets the list of apps from remote storage
+func (client *AzureBlobClient) GetAppsList(ctx context.Context) (RemoteDataListResponse, error) {
```

Comment:
one minor change which can be done is to use a variable like `var azureRemoteDataResponse RemoteDataListResponse{}` and return azureRemoteDataResponse in all the error and non-error conditions.

Reply:
sounds good. if you are ok, I can include this change in the coming spring work.